grid-demo/src/app/core/advanced-row-selection/advanced-row-selection.component.html

```diff
@@ -1 +1,62 @@
-Core - Advanced Row Selection
+<cds-grid
+  *ngIf="
+    {
+      vms: vms | async,
+      selectedMap: selectedMap | async,
+      selectedCount: selectedCount | async,
+      pageIndex: pageIndex | async,
+      pageCount: pageCount | async,
+      pageSize: pageSize | async,
+      entirePageSelected: entirePageSelected | async
+    };
+    let data
+  "
+  aria-label="basic datagrid demo"
+  height="400"
+>
+  <cds-grid-column type="action">
+    <cds-checkbox (click)="stopPropagation($event)">
+      <input (click)="selectAll()" type="checkbox" [checked]="data.entirePageSelected" aria-label="Select All" />
+    </cds-checkbox>
+  </cds-grid-column>
+  <cds-grid-column *ngFor="let column of columns">{{ column.displayName }}</cds-grid-column>
+
+  <ng-container *ngIf="data.vms === null">
+    <cds-grid-placeholder>
+      <cds-progress-circle size="xl" status="info"></cds-progress-circle>
+      <p cds-text="subsection">Loading Data</p>
+    </cds-grid-placeholder>
+  </ng-container>
+
+  <cds-grid-row
+    *ngFor="let vm of data.vms"
+    (click)="rowClicked($event, vm.id)"
+    [attr.selected]="data.selectedMap![vm.id]"
+  >
+    <cds-grid-cell>
+      <cds-checkbox (click)="stopPropagation($event)">
```

Comment:
This seems like a weird hack, hmm

Reply:
Yes, I'd love any insights you can give. If I use a regular input checkbox (not wrapped in a cds-checkbox), then everything works fine. When I use the cds-checkbox, I have to add this otherwise it won't work properly. Do you have any idea what's going on?

Reply:
I don't, I can take a look after it's merged in. Does it not work at all? Or is it flakey? I noticed some flakiness in the grid in the core storybook, that I hadn't investigated yet, but for the most part it worked fine in vanilla web components.

Reply:
I think it has something to do with the cds-checkbox and not the grid (just my guess). The behavior is consistent and the select-all doesn't work at all if I don't have the stopPropagation.